src/RateSys.jl

```diff
@@ -0,0 +1,53 @@
+#  |                    |                        |                  |
+# t_i    autonomous    t_ni   non-autonomous   t_nf   autonomous   t_f
+#  |                    |                        |                  |
+
+using DynamicalSystems
+using Plots
+
+# we write
+
+function RateSyst(tni,tnf,f,λ,p_λ,initvals)
```

Comment:
In my opinion, these are quite many positional arguments, which makes it less user-friendly. What do you think of reorganising this in the following structure:

```julia
RateSystem(ds::ContinuousTimeDynamicalSystem, rate_protocol::function)
```

where `ds` can be a `CoupledODEs` or a `CoupledSDEs` and `rate_protocol` is your `λ` function that determines how the parameters change over time. This function could have the structure

```julia
rate_protocol(u, p_lambda, t; t_start=0.0, t_end=Inf, kwargs...)
```
i.e. it would contain all the input arguments needed to define the rate forcing function. Behind the scenes, this would then create and return a new `CoupledODEs` or `CoupledSDEs` with the explicit time dependence specified by `rate_protocol`.

Reply:
The naming here is just a suggestion, but generally I think we should give more descriptive names than just `lambda` or `tnf` (clarity over brevity!)

Reply:
This sounds great, thanks Reyk! However, rate_protocol should not depend on u, in my opinion.